### PR TITLE
Add acceptance test for crashed app instance stats

### PR DESCRIPTION
**Pull request by kathap**
When pushing an app which has crashing instances (e.g. last instance crashes) all instances should be reported with their correct state.

### Are you submitting this PR against the [develop branch](https://github.com/cloudfoundry/cf-acceptance-tests/tree/develop)?

_All PR's to CATs should be submitted to develop and will be merged to main once they've passed acceptance._

### What is this change about?
Add a test when pushing an app which has crashing instances (e.g. last instance crashes) check that the states of the instances are reported correctly.

### Please provide contextual information.

[#3953](https://github.com/cloudfoundry/cloud_controller_ng/pull/3953)

### What version of cf-deployment have you run this cf-acceptance-test change against?

v42.3.0

### Please check all that apply for this PR:

- [ ] introduces a new test --- Are you sure everyone should be running this test?
- [X] changes an existing test
- [ ] requires an update to a CATs integration-config

### Did you update the README as appropriate for this change?

- [ ] YES
- [X] N/A

### If you are introducing a new acceptance test, what is your rationale for including it CATs rather than your own acceptance test suite?

_CATs should validate common operator workflows._
_CATs is not a regression test suite._
_CATs is run by every component team to validate their releases before promotion._

### How many more (or fewer) seconds of runtime will this change introduce to CATs?

~90 seconds

### What is the level of urgency for publishing this change?

- [ ] **Urgent** - unblocks current or future work
- [X] **Slightly Less than Urgent**

### Tag your pair, your PM, and/or team!
@johha @jochenehret

### CODE REVIEW

File: assets/python-crash-app/requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+# requirements.txt
```
Comment (jochenehret) [inline assets/python-crash-app/requirements.txt:1]: Do we really need this empty file?

File: apps/crashing.go [hunk 2]
```diff
@@ -45,6 +53,69 @@ var _ = AppsDescribe("Crashing", func() {
 		})
 	})
 
+	Describe("an app with three instances, two running and one crashing", func() {
+		It("keeps two instances running while another crashes", func() {
+			By("Pushing the app with three instances")
+			Expect(cf.Cf(
+				"push", appName,
```
Comment (jochenehret) [inline apps/crashing.go:60]: Some parameters are already defined in the test app's manifest, do we need both?

Comment (kathap) [reply apps/crashing.go:60]: Not really..

File: apps/crashing.go [hunk 3]
```diff
@@ -45,6 +47,69 @@ var _ = AppsDescribe("Crashing", func() {
...
+			By("Checking that the app is up and running")
+			Eventually(cf.Cf("app", appName)).Should(Say("running"))
+
+			By("Waiting until one instance crashes")
+			appGuid := app_helpers.GetAppGuid(appName)
+			processStatsPath := fmt.Sprintf("/v3/apps/%s/processes/web/stats", appGuid)
+
+			// Poll until at least one instance has crashed
+			Eventually(func() bool {
+				session := cf.Cf("curl", processStatsPath).Wait()
```
Comment (jochenehret) [inline apps/crashing.go:70]: Could this block be extracted into a function? Then you could call
```
hasOneInstanceState("CRASHED")
...
hasOneInstanceState("RUNNING")
```

Comment (kathap) [reply apps/crashing.go]: Good catch

**Review by johha — DISMISSED** [review]

File: apps/crashing.go [hunk 4]
```diff
@@ -1,19 +1,47 @@
...
 	"github.com/cloudfoundry/cf-test-helpers/v2/cf"
 	"github.com/cloudfoundry/cf-test-helpers/v2/helpers"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gbytes"
+	. "github.com/onsi/gomega/gexec"
+	"time"
 )
 
+func hasOneInstanceInState(appName, processPath, desiredState string) bool {
```
Comment (jochenehret) [inline apps/crashing.go:19]: "appName" parameter is not used in function?

**Review by jochenehret — CHANGES_REQUESTED** [review]

**Review by jochenehret — APPROVED** [review]

**Review by johha — APPROVED** [review]

**Pull request merged**